genericache/__init__.py

```
@@ -109,26 +135,26 @@ def try_fetch(
         url: U,
         fetcher: "Callable[[U], Iterable[bytes]]",
         force_refetch: "bool | ContentDigest",
-    ) -> "CacheEntry | FetchInterrupted[U] | ContentUnavailable":
-        ...
+    ) -> "CacheEntry | FetchInterrupted[U] | DigestMismatch[U]": ...
 
     def fetch(
         self,
         url: U,
         fetcher: "Callable[[U], Iterable[bytes]]",
         force_refetch: "bool | ContentDigest" = False,
-        retries: int = 3
+        retries: int = 3,
     ) -> "CacheEntry":
         for _ in range(retries):
             result = self.try_fetch(url, fetcher, force_refetch=force_refetch)
             if isinstance(result, CacheEntry):
                 return result
             if isinstance(result, FetchInterrupted):
                 continue
-            if isinstance(result, Exception): # pyright: ignore[reportUnnecessaryIsInstance]
-                raise result
+            raise result
```

I guess I would keep the `isinstance`, but check explicitly for `DigestMismatch` just so that we know exactly which exception we're gonna handle and which we're gonna raise. If we ever change the signature of `try_fetch`, I wouldn't want any extra exceptions to be blindly swallowed by the `raise`.